scripts/multisig/change_service_multisig_owners_data.js

```diff
@@ -0,0 +1,135 @@
+/*global process*/
+
+const { ethers } = require("hardhat");
+
+async function main() {
+    // Get the multisig data from file
+    const fs = require("fs");
+    // "multisig_info.json" test file is located in the same folder of this script
+    const dataJSON = "multisig_info.json";
+    const dataFromJSON = fs.readFileSync(dataJSON, "utf8");
+    const parsedFile = JSON.parse(dataFromJSON);
+
+    // Set of updated service multisig owners
+    const agentInstances = parsedFile["agentInstances"];
+    // Service multisig address
+    const serviceMultisigAddress = parsedFile["serviceMultisigAddress"];
+    // Service owner address (current multisig owner)
+    const serviceOwnerAddress = parsedFile["serviceOwnerAddress"];
+    // Safe MultisendCallOnly contract address
+    const multisendAddress = parsedFile["multisendAddress"];
+    // Resulting threshold
+    const threshold = parsedFile["threshold"];
+    // Current service multisig nonce
+    const nonce = parsedFile["nonce"];
+    // Chain Id
+    const chainId = parsedFile["chainId"];
+    // If true, the safeTx hash is already approved in the service multisig
+    // If false, the safeTx hash needs to be signed by the service multisig owner
+    const approvedHash = parsedFile["approvedHash"];
+    const sentinelOwners = "0x" + "0".repeat(39) + "1";
+
+    // Sanity check
+    if (!agentInstances || agentInstances.length == 0 || !serviceMultisigAddress || !serviceOwnerAddress ||
+        !multisendAddress || !threshold || !nonce || !chainId) {
+        return;
+    }
+
+    // Get the Safe master contract and multiSend one for ABI purposes
+    const GnosisSafe = await ethers.getContractFactory("GnosisSafe");
+    const gnosisSafe = await GnosisSafe.deploy();
+    await gnosisSafe.deployed();
+
+    const MultiSend = await ethers.getContractFactory("MultiSendCallOnly");
+    const multiSend = await MultiSend.deploy();
+    await multiSend.deployed();
+
+    const safeContracts = require("@gnosis.pm/safe-contracts");
+
+    // Change existent multisig owners and a threshold in a multisend transaction using the service owner access
```

Comment:
This can be used when the previous agent instances already gave up their multisig ownership to the service owner. What about the calldata  that allows such agent instances to give the ownership rights to the service owner with a multisend transaction using the agents' instances rights?

Reply:
The downgrading was not part of that script. I do not believe it was requested (yet).

Reply:
Currently, we have a multisig with 4 owners as far I can tell. So I guess it is necessary to go first from agent instances to service owner, no?

Reply:
Currently services start with 1 agent instance, and then upgrade to 4 or more.

Reply:
We have concluded that the downgrading is not part of the task (yet).